src/libs/DebugUtils.ts

```diff
@@ -419,88 +424,425 @@ function validateString(value: string) {
     }
 }
 
+/**
+ * Execute validation of a union type (e.g. Record<string, string> | Array<string>)
+ */
+function unionValidation(firstValidation: () => void, secondValidation: () => void) {
+    try {
+        firstValidation();
+    } catch (e) {
+        secondValidation();
+    }
+}
+
 /**
  * Validates if a property of Report is of the expected type
  *
  * @param key - property key
  * @param value - value provided by the user
  */
 function validateReportDraftProperty(key: keyof Report, value: string) {
-    if (REPORT_REQUIRED_PROPERTIES.includes(key) && value === 'undefined') {
+    if (REPORT_REQUIRED_PROPERTIES.includes(key) && isEmptyValue(value)) {
         throw SyntaxError('debug.missingValue');
     }
-    if (key === 'privateNotes') {
-        return validateObject(
-            value,
-            {
-                note: 'string',
-            },
-            'number',
-        );
-    }
-    if (key === 'permissions') {
-        return validateArray(value, CONST.REPORT.PERMISSIONS);
-    }
-    if (key === 'pendingChatMembers') {
-        return validateArray(value, {
-            accountID: 'string',
-            pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION,
-        });
-    }
-    if (key === 'participants') {
-        return validateObject(
-            value,
-            {
-                notificationPreference: CONST.REPORT.NOTIFICATION_PREFERENCE,
-            },
-            'number',
-        );
-    }
-    if (REPORT_NUMBER_PROPERTIES.includes(key)) {
-        return validateNumber(value);
-    }
-    if (REPORT_BOOLEAN_PROPERTIES.includes(key)) {
-        return validateBoolean(value);
-    }
-    if (REPORT_DATE_PROPERTIES.includes(key)) {
-        return validateDate(value);
-    }
-    if (key === 'tripData') {
-        return validateObject(value, {
-            startDate: 'string',
-            endDate: 'string',
-            tripID: 'string',
-        });
-    }
-    if (key === 'lastActionType') {
-        return validateConstantEnum(value, CONST.REPORT.ACTIONS.TYPE);
-    }
-    if (key === 'writeCapability') {
-        return validateConstantEnum(value, CONST.REPORT.WRITE_CAPABILITIES);
-    }
-    if (key === 'visibility') {
-        return validateConstantEnum(value, CONST.REPORT.VISIBILITY);
-    }
-    if (key === 'stateNum') {
-        return validateConstantEnum(value, CONST.REPORT.STATE_NUM);
-    }
-    if (key === 'statusNum') {
-        return validateConstantEnum(value, CONST.REPORT.STATUS_NUM);
-    }
-    if (key === 'chatType') {
-        return validateConstantEnum(value, CONST.REPORT.CHAT_TYPE);
-    }
-    if (key === 'errorFields') {
-        return validateObject(value, {});
-    }
-    if (key === 'pendingFields') {
-        return validateObject(value, {});
-    }
-    if (key === 'participantAccountIDs') {
-        return validateArray(value, 'number');
+    switch (key) {
```

Comment:
Is possible to simplify this? For example, combining all string validations in one case:

```tsx
switch (key) {
        // all string cases
        case 'avatarUrl':
        case 'avatarFileName':
        case 'lastVisibleActionCreated':
        case 'lastReadCreated':
        case 'lastReadTime': {
            return validateString(value);
        }
```